packages/hoppscotch-common/src/services/banner.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Service } from "dioc"
+import { ref } from "vue"
+import { getI18n } from "~/modules/i18n"
+
+export type BannerType = "info" | "warning" | "error"
+
+export type BannerContent = {
+  type: BannerType
+  text: (t: ReturnType<typeof getI18n>) => string
```

Comment:
i see now we're enforcing the use of i18n for the text, but that might not be the best idea, because as far as the banner is concerned, it's just a string. it is the responsibility of the component using this banner to decide whether to use i18n or not.

i would love to hear your points if you think using i18n here is better, 

```suggestion
  text: string
```

Reply:
I think the suggested approach is better.